netkan/netkan/scheduler.py

```diff
@@ -2,14 +2,48 @@
 import datetime
 import logging
 import requests
+import json
+import re
 from pathlib import Path
 from hashlib import md5
 
 
+class Netkan:
+
+    kref_pattern = re.compile('^#/ckan/([^/]+)/(.+)$')
+
+    def __init__(self, filename):
+        self.filename = Path(filename)
+        self.contents = self.filename.read_text()
+        self.json     = json.loads(self.contents)
```

Comment:
I don't think there is a reason to store the contents of the file in memory, we can likely pass that straight to json.loads(). The `self.json` should really be called `raw` or `_raw` as it has been loaded in as a python object. The underscore is regularly used to identify it as private to the class, but it's just there as a 'YOYO' if you access this outside the class and expect it not to break.

Reply:
`contents` is used below twice in `sqs_message`.

Reply:
That's a point. Feels like a lot of duplication of data happening, in the scheme of things it probably doesn't matter.